api/src/jobs/Ingest.test.ts

```diff
@@ -0,0 +1,89 @@
+import Config from "../models/Config";
+import Job from "../models/Job";
+import { FedoraObject } from "../models/FedoraObject";
+import Fedora from "../services/Fedora";
+import FedoraObjectFactory from "../services/FedoraObjectFactory";
+import { IngestProcessor } from "./Ingest";
+import QueueManager from "../services/QueueManager";
+import fs = require("fs");
+import winston = require("winston");
+
+// We have an indirect dependency on ImageFile, but we don't really want
+// to load it for the context of this test.
+jest.mock("../models/ImageFile.ts", () => {
+    return {};
+});
+
+describe("IngestProcessor", () => {
+    let ingest: IngestProcessor;
+    let config: Config;
+    let logger: winston.Logger;
+    let dir: string;
+    let jobName: string;
+    let fedora: Fedora;
+    let job: Job;
+    beforeEach(() => {
+        dir = "/my/fake/dir";
+        jobName = "fakejob";
+        config = new Config({ processed_area_path: "/fake_processed" });
+        logger = winston.createLogger({
+            level: "error", // we don't want to see info messages while testing
+            transports: [new winston.transports.Console()],
+        });
+        fedora = new Fedora(config);
+        job = new Job(dir + "/" + jobName, config, new QueueManager());
+        jest.spyOn(Job, "build").mockReturnValue(job);
+        ingest = new IngestProcessor(dir, config, new FedoraObjectFactory(config), logger);
+    });
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+    describe("finalizeTitle", () => {
+        it("creates a title based on the directory path", async () => {
+            const fedoraObject = new FedoraObject("foo:123", config, fedora);
+            const labelSpy = jest.spyOn(fedoraObject, "modifyObjectLabel").mockImplementation(jest.fn());
+            const datastreamSpy = jest
+                .spyOn(fedoraObject, "getDatastream")
+                .mockResolvedValue("<oai:dc><dc:title>Incomplete... / Processing...</dc:title></oai:dc>");
+            const replaceSpy = jest.spyOn(ingest, "replaceDCMetadata").mockImplementation(jest.fn());
+            await ingest.finalizeTitle(fedoraObject);
+            expect(labelSpy).toHaveBeenCalledTimes(1);
+            expect(labelSpy).toHaveBeenCalledWith("fakejob_dir_fake_my");
+            expect(datastreamSpy).toHaveBeenCalledTimes(1);
+            expect(datastreamSpy).toHaveBeenCalledWith("DC");
+            expect(replaceSpy).toHaveBeenCalledTimes(1);
+            expect(replaceSpy).toHaveBeenCalledWith(
+                fedoraObject,
+                "<oai:dc><dc:title>fakejob_dir_fake_my</dc:title></oai:dc>",
+                "Set dc:title to ingest/process path"
+            );
+        });
+    });
+    describe("moveDirectory", () => {
+        it("moves the directory appropriately", () => {
+            const date = new Date();
```

Comment:
I think a better solution is actually to use a pattern matcher on the expectation for the filename, because I want to be sure there's a date in the resulting output, but I don't really care what date it is. I wasn't able to figure out how to do that before, so I did this out of laziness, but I think I have an idea of how to improve it... I'll try momentarily.

Reply:
Okay, see [cb83ec5](https://github.com/FalveyLibraryTechnology/VuDLPrepJS/pull/117/commits/cb83ec5314b1bcbb12a2613be01387cbc452d802) -- I think that's much nicer. (I could probably make my regular expression more concise, but I had trouble getting the code to recognize the `{n}` syntax for repeating patterns, and rather than waste a lot of time troubleshooting, I just made a somewhat repetitive regex).